boards: nrfbsim: Fix testargs with non host libC

aescolar opened the pull request:
Ensure that we call the host libC when allocating space for a possible set of test arguments even if we are building Zephyr with an embedded libC.

This fixes a bug, where a test which uses the bsim test args facility and builds with an embedded libC instead of the host libC would fail at runtime.

---

Added an extra commit to add a filter for the test which caused twister to crash.
In short: when building images which use IPC for the nrf5340bsim//cpunet the build will (purposely) only build that core image elf, but not link the final executable. This is because to build the final executable one needs the app core image (which contains the IPC buffer).
twister when building for native targets expects the final exe to run the test, and if it does not find it, it dies badly. Let's add a filter to ensure this test only runs for native targets if the exe has been produced. For non posix arch targets the filter is always true.

aescolar committed:
boards: nrfbsim: Fix testargs with non host libC

Ensure that we call the host libC when allocating space
for a possible set of test arguments even if we are
building Zephyr with an embedded libC.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

koffes reviewed (approved):
Thanks for the fast fix! Tested and works great.

aescolar committed:
tests/subsys/ipc/pbuf: Filter out partial native/posix arch builds

Some AMP native simulator targets (like the nrf5340bsim net core)
will not produce a final executable if only one of the core images
is built, when IPC is enabled, as those images require the other
core image to be able to link.
So this test cannot be run when targetting that core alone
(without using sysbuild to build the other).

Let's filter out these targets to avoid twister building
but finding itself with no executable and error out badly
as it cannot run the test.

The filter condition is effectively:
If building for a native target, there must be an .exe being
produced.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>
